listenbrainz/bigquery-writer/bigquery-writer.py

```diff
@@ -81,7 +81,7 @@ def callback(self, ch, method, body):
             }
             bq_data.append({
                 'json': row,
-                'insertId': "%s-%s" % (listen['user_name'], listen['listened_at'])
+                'insertId': "%s-%s-%s" % (listen['user_name'], listen['listened_at'], listen['recording_msid'])
```

Comment:
testing against external services is difficult. We could try and create some sort of 'fake bq' local service, but we would have to replicate behaviour that we don't understand in order to test this, which is almost impossible.

In this case, perhaps we could split the part that generates `bq_data` into a separate method and then have a test for it which checks that we have unique `insertId`s for all items.
However, I'm not sure how useful this check will be - it's not going to help us find a problem in bq until we actually observe the error happening there...